app/src/main/kotlin/com/wire/android/ui/conversation/common/EventBadge.kt

```diff
@@ -0,0 +1,93 @@
+package com.wire.android.ui.conversation.common
+
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.wrapContentWidth
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.wire.android.R
+import com.wire.android.ui.conversation.all.model.EventType
+import com.wire.android.ui.theme.WireColor
+
+
+@Composable
+fun EventBadgeFactory(eventType: EventType, modifier: Modifier = Modifier) {
+    when (eventType) {
+        EventType.MissedCall -> MissedCallBadge(modifier)
+        EventType.UnreadMention -> UnreadMentionBadge(modifier)
+        is EventType.UnreadMessage -> UnreadMessageEventBadge(unreadMessageCount = eventType.unreadMessageCount, modifier)
+        EventType.UnreadReply -> UnreadReplyBadge(modifier)
+    }
+}
+
+@Composable
+fun MissedCallBadge(modifier: Modifier = Modifier) {
+    Image(
+        painter = painterResource(id = R.drawable.ic_event_badge_missed_call),
+        contentDescription = null,
+        modifier = modifier
+    )
+}
+
+@Composable
+fun UnreadMentionBadge(modifier: Modifier = Modifier) {
+    Image(
+        painter = painterResource(id = R.drawable.ic_event_badge_unread_mention),
+        contentDescription = null,
+        modifier = modifier
+    )
+}
+
+@Composable
+fun UnreadReplyBadge(modifier: Modifier = Modifier) {
+    Image(
+        painter = painterResource(id = R.drawable.ic_event_badge_unread_reply),
+        contentDescription = null,
+        modifier = modifier
+    )
+}
+
+@Composable
+fun UnreadMessageEventBadge(unreadMessageCount: Int, modifier: Modifier = Modifier) {
+    if (unreadMessageCount > 0) {
+        Text(
+            text = unReadMessageCountStringify(unreadMessageCount),
+            color = Color.White,
+            style = TextStyle(
+                textAlign = TextAlign.Center,
+                fontWeight = FontWeight.W700,
+                letterSpacing = 0.25.sp,
+                fontStyle = FontStyle.Normal,
+                lineHeight = 14.sp
+            ),
+            modifier = modifier
+                .wrapContentWidth()
+                .background(
+                    color = WireColor.Dark90Gray,
```

Comment:
We shouldn't refer directly to WireColor, we should use MaterialTheme.colors.X because this way our theme works correctly and we get the proper light/dark version colors - if we want more colors than it has by default then we have to create our own custom Colors palette that extends the default one.

It might stay that way for now, I was thinking about sorting it out soon, while there aren't too many of these references yet. I wanted to know if someone has any opinion on this subject?

If we want to do it the Material way, then we should pass the "gray" color as a custom one, and assume that all shades like "gray90" are just "gray" with some alpha thus we should use `ContentAlpha`, but I'm not sure if our designs are prepared this way  so another approach is to prepare some custom colors like "badgeColor" and set proper values in our light/dark palettes.

Reply:
I think this could an topic to discuss indeed